frontend/src/components/ExternalResourceModal.js

```diff
@@ -60,12 +56,18 @@ const ExternalLink = ({ to, children }) => {
   }, [onEscape]);
 
   useEffect(() => {
+    if (!modalRef.current) return;
```

Comment:
Due to moving the return statement to the bottom, `modalRef` may be null, since we may never render the modal. Add a check to make sure we skip if we didn't render it.